File: fortran/initialize_activity.f90
            subroutine initialize_activity
            use all_parameters
            implicit none
            integer::i
            double precision :: aaa,ran1
			double precision x, y, z
            !!!!!!!FIND THE DIRECTION OF THE ACTIVE VELOCITY
            do i=1,no_of_colloid
              ra(3*i-2)= pos_colloid(3*i-2)-ran1(zzzz)*llx
              ra(3*i-1)= pos_colloid(3*i-1)-ran1(zzzz)*lly
              ra(3*i)= pos_colloid(3*i)-ran1(zzzz)*llz
              x = ra(3*i - 2)
			  y = ra(3*i - 1)
			  z = ra(3*i)
              aaa=dsqrt(x**2+y**2+z**2)

              ra(3*i-2)=ra(3*i-2)/aaa !RA IS THE UNIT VECTOR TOWARDS THE ACTIVE VELOCITY
              ra(3*i-1)=ra(3*i-1)/aaa
              ra(3*i)=ra(3*i)/aaa

              vel_colloid(3*i-2)=vel_colloid(3*i-2)+ra(3*i-2)*v0
              vel_colloid(3*i-1)=vel_colloid(3*i-1)+ra(3*i-1)*v0
              vel_colloid(3*i)=vel_colloid(3*i)+ra(3*i)*v0

            enddo
            end subroutine initialize_activity

File: cpp/initialize_activity.cpp
#include "parameters.hpp"
#include <cstdlib>

void initialize_activity() {
	double aaa; //seriously better names
	for(int i = 1; i <= no_of_colloid; i++) {
		ra[3*i - 2] = pos_colloid[3*i - 2] - rand()*llx; 
//And ran1--- variable or function?
		ra[3*i - 1] = pos_colloid[3*i - 1] - rand()*lly;
		ra[3*i] = pos_colloid[3*i] - rand()*llx;
		aaa = sqrt(pow(ra[3*i - 2],2) + pow(ra[3*i -1], 2) + pow(ra[3*i],2));

		for(int j = 0; j <= 2; j++) {
			ra[3*i - j] /= aaa;
			vel_colloid[3*i - j] += ra[3*i - j]*v0;
		}
	}
}
